protocol/src/main/java/org/mifos/chatbot/protocol/telegram/TelegramChatService.java

```diff
@@ -0,0 +1,183 @@
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+
+@Slf4j
+@Component
+public class TelegramChatService extends TelegramLongPollingBot {
+
+    @Autowired
+    private MifosSettings settings;
+
+    @Autowired
+    private AdapterService adapterService;
+
+    @Autowired
+    DataSource dataSource;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Value("${telegram.botUsername}")
+    private String botUsername;
+
+    @Value("${telegram.botToken}")
+    private String botToken;
+
+    @Override
+    public void onUpdateReceived(Update update) {
+        if (update.hasMessage() && update.getMessage().hasText()) {
+            final String messageText = update.getMessage().getText();
+            final long senderId = update.getMessage().getChatId();
+            log.info("Telegram: Message received of length " + messageText.length() + " from user: " + senderId);
+            if (messageText.toLowerCase().contains("logout")) {
+                User user = userRepository.findUserByTelegramID(Long.toString(senderId));
+                if(user != null) {
+                    userRepository.removeUser(user.getUsername());
+                    sendTextMessage(senderId, "Logged out successfully.");
+                } else {
+                    sendTextMessage(senderId, "You are already logged out.");
+                }
+                return;
+            }
+            if (messageText.toLowerCase().contains("login:")) {
+                StringBuilder username = new StringBuilder();
+                StringBuilder password = new StringBuilder();
+                String creds = messageText.replaceAll("login:", "");
+                for (int i = 0; i < creds.length(); i++) {
+                    if (creds.charAt(i) == ':') {
+                        for (i++; i < creds.length(); i++) {
+                            password.append(creds.charAt(i));
+                        }
+                        break;
+                    }
+                    username.append(creds.charAt(i));
+                }
+                if(userRepository.telegramIDExist(Long.toString(senderId))) {
+                    sendTextMessage(senderId, "You are already logged in, please logout first.");
+                } else {
+                    if (authUser(username.toString(), password.toString())) {
+                        User user = userRepository.findUserByUsername(username.toString());
+                        if (user == null) {
+                            userRepository.addUserByTelegramID(
+                                    username.toString(),
+                                    password.toString(),
+                                    Long.toString(senderId)
+                            );
+                        } else {
+                            userRepository.updateUserTelegramID(
+                                    username.toString(),
+                                    password.toString(),
```

Comment:
Ideally we should encode passwords

Reply:
Encoded to base64 as currently i don't have any solution to store them by one way encryption as i need them to do API calls (demo.openmf.org)